common/lib/xmodule/xmodule/license.py

```diff
@@ -0,0 +1,168 @@
+
+    enforce_type = from_json
+
+
+class ARRLicense(License):
+    """
+    License class for an 'All rights reserved' license
+    """
+
+    def __init__(self, kind="ARR", version=None, *args, **kwargs):
+        super(ARRLicense, self).__init__(kind, version, *args, **kwargs)
+
+    @property
+    def html(self):
+        """
+        Return a piece of html that descripts the license
+        """
+        phrase = _("All rights reserved")
+        return "&copy;<span class='license-text'>{phrase}</span>".format(
+            phrase=phrase
+        )
+
+
+class CCLicense(License):
+    """
+    License class for a Creative Commons license
+    """
+
+    def __init__(self, kind, version=None, *args, **kwargs):
+        super(CCLicense, self).__init__(kind, version, *args, **kwargs)
+        # If no version was set during initialization, we may assume
+        # the most recent version of a CC license and fetch
+        # that using the API.
+        if self.kind and not self.version:
+            self.version = "4.0"
+
+    @property
+    def html(self):
+        """
+        Return a piece of html that describes the license
+        """
+
+        license_html = ["<i class='icon-cc'></i>"]
+        if 'CC0' in self.kind:
+            license_html.append("<i class='icon-cc-zero'></i>")
+        if 'BY' in self.kind:
+            license_html.append("<i class='icon-cc-by'></i>")
+        if 'NC' in self.kind:
+            license_html.append("<i class='icon-cc-nc'></i>")
+        if 'SA' in self.kind:
+            license_html.append("<i class='icon-cc-sa'></i>")
+        if 'ND' in self.kind:
+            license_html.append("<i class='icon-cc-nd'></i>")
+
+        phrase = _("Some rights reserved")
+        return "<a rel='license' href='http://creativecommons.org/licenses/{license_link}/{version}/' " \
+            "data-tooltip='{description}' target='_blank' class='license'>" \
+            "{license_html}<span class='license-text'>{phrase}</span></a>".format(
+                description=self.description,
+                version=self.version,
+                license_link=self.kind.lower()[3:],
+                license_html=''.join(license_html),
+                phrase=phrase
+            )
+
+    @property
+    def description(self):
+        """
+        Return a text that describes the license
+        """
+        cc_attributes = []
+        if 'BY' in self.kind:
+            cc_attributes.append(_("Attribution"))
+        if 'NC' in self.kind:
+            cc_attributes.append(_("NonCommercial"))
+        if 'SA' in self.kind:
+            cc_attributes.append(_("ShareAlike"))
+        if 'ND' in self.kind:
+            cc_attributes.append(_("NonDerivatives"))
+
+        return _("This work is licensed under a Creative Commons {attributes} {version} International License.").format(
+            attributes='-'.join(cc_attributes),
+            version=self.version
+        )
+
+def parse_license(kind_or_license, version=None):
```

Comment:
This whole method seems like a bad idea. Why do we want to allow licenses to exist in any of several different representations or states? Seems to me like we should expect a license to be represented in exactly one way.